lib/optimizely/optimizely_config.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+#    Copyright 2019, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+module Optimizely
+  class OptimizelyConfig
```

Comment:
This is not a blocker as discussed offline, I would suggest that we should persist OptimizelyConfig response in this class. And each time config is called, it should expect a project config and we only generate new maps if the project ID  or revision has changed, otherwise we can return the same response. 

Datafile is unlikely to change so often. And we can save time from unnecessarily creating and merging so many maps to generate the same response again and again. Especially when the datafile is huge with many features/ experiments/ variables.

Reply:
Its a great suggestion and i remember we discussed it offline as well. I did it this way because its explicitly mentioned in the documentation that a new object will be generated every time. But i agree, this makes more sense to me as well. 
@jaeopt @mjc1283 .. What do you guys think about storing optimizelyconfig and creating a new one only if project config has a newer revision?

Reply:
@zashraf1985 @oakbani both Jae and Matt agree that we should add caching fo the OptimizelyConfig.  Can you update the PR?

Reply:
@thomaszurkan-optimizely You are abolutely right. But its also decided to do it in two separate PRs and merge them one by one in to master. So this should be considered as the first PR without caching. Once this is merged, we will create another one for caching.